Weird error with new signals lifetimes

**Problem**

I am getting an error with signals that seems unnecessary and it used to compile just fine before.

Code:
```rs
#[allow(non_snake_case)]
#[component]
fn Comp(value: i32) -> Element {
    rsx!(
        label {
            "{value}"
        }
    )
}

fn app() -> Element {
    let count = use_signal(|| 0);

    rsx!(
        Comp {
            value: *count.read()
        }
    )
}
```
Error:
```sh
> cargo run --example counter
    Blocking waiting for file lock on build directory
   Compiling examples v0.0.0 (D:\Projects\freya)
error[E0597]: `count` does not live long enough
  --> examples\counter.rs:27:21
   |
23 |     let count = use_signal(|| 0);
   |         ----- binding `count` declared here
...
27 |             value: *count.read()
   |                     ^^^^^-------
   |                     |
   |                     borrowed value does not live long enough
   |                     a temporary with access to the borrow is created here ...
...
30 | }
   | -
   | |
   | `count` dropped here while still borrowed
   | ... and the borrow might be used here, when that temporary is dropped and runs the destructor for type `generational_box::references::GenerationalRef<Ref<'_, i32>>`
   |
   = note: the temporary is part of an expression at the end of a block;
           consider forcing this temporary to be dropped sooner, before the block's local variables are dropped
help: for example, you could save the expression's value in a new local variable `x` and then make `x` be the expression at the end of the block
   |
25 ~     let x = rsx!(
26 |         Comp {
27 |             value: *count.read()
28 |         }
29 ~     ); x
   |

For more information about this error, try `rustc --explain E0597`.
error: could not compile `examples` (example "counter") due to previous error
```

**Expected behavior**

Compile

**Environment:**
 - Dioxus version: `2d2e9dc56a59b4f3ad4deb55d15c27f337cf564c`
 - Rust version: 1.76
 - OS info: Windows
 - App platform: `freya`

--- a/packages/hooks/src/use_resource.rs
+++ b/packages/hooks/src/use_resource.rs
@@ -107,6 +107,13 @@ pub struct Resource<T: 'static> {
     callback: UseCallback<Task>,
 }
 
+impl<T> Clone for Resource<T> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+impl<T> Copy for Resource<T> {}
+
 /// A signal that represents the state of the resource
 // we might add more states (panicked, etc)
 #[derive(Clone, Copy, PartialEq, Hash, Eq, Debug)]

--- a/packages/rsx/src/lib.rs
+++ b/packages/rsx/src/lib.rs
@@ -244,12 +244,16 @@ impl<'a> ToTokens for TemplateRenderer<'a> {
                 attr_paths: &[ #(#attr_paths),* ],
             };
 
-            dioxus_core::VNode::new(
-                #key_tokens,
-                TEMPLATE,
-                Box::new([ #( #node_printer),* ]),
-                Box::new([ #(#dyn_attr_printer),* ]),
-            )
+            {
+                // NOTE: Allocating a temporary is important to make reads within rsx drop before the value is returned
+                let __vnodes = dioxus_core::VNode::new(
+                    #key_tokens,
+                    TEMPLATE,
+                    Box::new([ #( #node_printer),* ]),
+                    Box::new([ #(#dyn_attr_printer),* ]),
+                );
+                __vnodes
+            }
         });
     }
 }

